content/shmem_malloc_hints.tex

```diff
@@ -44,6 +44,17 @@
     The user is also responsible for calling these routines with identical argument(s) on all
     \acp{PE}; if differing \VAR{size}, or \VAR{hints} arguments are used, the behavior of the call
     and any subsequent \openshmem calls is undefined.
+
+	Calling the routine \FUNC{shmem\_realloc(void *ptr, size\_t size)} on the
+	ptr allocated by \FUNC{shmem\_malloc\_with\_hints} has a behavior similar to
+	calling \FUNC{shmem\_realloc} on the ptr allocated by \FUNC{shmem\_malloc}.
+	If the new size is larger, the memory is allocated with the same hints that
+	were passed when calling \FUNC{shmem\_malloc\_with\_hints}, and the newly
```

Comment:
What about if the new size is smaller?

Reply:
Everything below this comment is duplicated from the `shmem_realloc` section. I think I am in favor of only keeping the first two sentences and moving them to where realloc is specified.

Reply:
I think we can add to the end of the `shmem_realloc` paragraph. I tried consider the case where realloc is used several times on the same buffer:
> If \VAR{ptr} corresponds to memory initially allocated by \FUNC{shmem\_malloc\_with\_hints}, the resulting memory will comply with the hints used in the initial allocation.

Reply:
> What about if the new size is smaller?

I did not define that behavior since we don't define it for shmem_realloc when ptr is allocated via shmem_malloc. I did not want to introduce new semantics. 

However,  it makes sense to define it as "shrink the size of memory". This change should be for shmem_realloc in general, not just wrt memory allocated via shmem_malloc_with_hints.